File: hw/vizcaino/u3/hw18OOPFarm/hw18Java/OOPConceptsFarm/src/main/java/ec/edu/espe/oopconceptsfarm/model/Pig.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.oopconceptsfarm.model;

/**
 *
 * @author Adrian Vizcaino <The-Softwarriors at ESPE>
 */
import java.util.ArrayList;
import java.util.Date;

public class Pig extends FarmAnimal implements IMeatAnimal {

    private float idealWeight;
    private int penNumber;
    private float totalMeatProduced;

    public Pig(int id, String breed,
            Date bornOn,
            float weight,
            String healthStatus,
            float idealWeight,
            int penNumber) {

        super(id, breed, bornOn, weight, healthStatus);

        this.idealWeight = idealWeight;
        this.penNumber = penNumber;
        this.totalMeatProduced = 0;
    }
    public void addMeatProduction(float kilograms){
    totalMeatProduced += kilograms;
}

public float getTotalMeatProduced(){
    return totalMeatProduced;
}

    @Override
    public ArrayList<Cut> cut() {

        ArrayList<Cut> cuts = new ArrayList<>();

        float meatYield;

        if (weight >= idealWeight) {

            meatYield = weight * 0.70f;

            cuts.add(new Cut(
                    1,
                    "Premium Pork",
                    "Premium Process",
                    meatYield));

        } else {

            meatYield = weight * 0.55f;

            cuts.add(new Cut(
                    1,
                    "Commercial Pork",
                    "Commercial Process",
                    meatYield));

        }

        return cuts;
    }

    @Override
    public void sendToSlaughterHouse(
            SlaughterHouse slaughterHouse) {

        System.out.println(
                "Pig sent to "
                + slaughterHouse.getName());
    }
}

File: hw/vizcaino/u3/hw18OOPFarm/hw18Python/model/pig.py
from model.farm_animal import FarmAnimal
from model.meat_animal import IMeatAnimal


class Pig(FarmAnimal, IMeatAnimal):
    def __init__(self, id, breed, born_on, weight, health_status, ideal_weight, pen_number):
        super().__init__(id, breed, born_on, weight, health_status)
        self.ideal_weight = ideal_weight
        self.pen_number = pen_number

    def cut(self):
        if self.weight >= self.ideal_weight:
            return {"type": "Premium", "yield": 0.70}
        return {"type": "Commercial", "yield": 0.55}

    def send_to_slaughter_house(self, slaughter_house):
        slaughter_house.receive(self)

    def to_dict(self):
        return self.__dict__